cpp/src/binaryop/compiled/Greater.cu

```diff
@@ -14,13 +14,22 @@
  * limitations under the License.
  */
 
-#include "binary_ops.cuh"
+#include <binaryop/compiled/binary_ops.cuh>
+#include <binaryop/compiled/struct_binary_ops.cuh>
 
 namespace cudf::binops::compiled {
-template void apply_binary_op<ops::Greater>(mutable_column_device_view&,
-                                            column_device_view const&,
-                                            column_device_view const&,
-                                            bool is_lhs_scalar,
-                                            bool is_rhs_scalar,
-                                            rmm::cuda_stream_view);
+template <>
+void apply_binary_op<ops::Greater>(mutable_column_view& out,
+                                   column_view const& lhs,
+                                   column_view const& rhs,
+                                   bool is_lhs_scalar,
+                                   bool is_rhs_scalar,
+                                   rmm::cuda_stream_view stream)
+{
+  is_struct(lhs.type()) && is_struct(rhs.type())
+    ? detail::struct_lexicographic_compare(
+        out, lhs, rhs, is_lhs_scalar, is_rhs_scalar, order::DESCENDING, false, stream)
+    : detail::apply_unnested_binary_op<ops::Greater>(
+        out, lhs, rhs, is_lhs_scalar, is_rhs_scalar, stream);
```

Comment:
Why does this dispatch to struct vs unnested binop need to be repeated in so many places instead of in a single place? i.e., this specialization should happen in the type_dispatch step.

Reply:
I pushed it level lower so the code is common. There is an operator dispatch that could be used as the fracture but that would result in a ternary operator for every case. There only type dispatch is the `device_type_dispatcher` in `binary_ops.cuh` which is on the `BinaryOperator` rather than the column type and happens too late for struct specific operations.